spec/mongoid/extensions/big_decimal_spec.rb

```diff
@@ -8,401 +8,853 @@
       end
 
-      let(:demongoized) do
-        BigDecimal.demongoize(neg_infinity)
+      context "when the value is BigDecimal NaN" do
+
+        let(:nan) do
+          BigDecimal("NaN")
+        end
+
+        it "returns a String" do
+          expect(BigDecimal.mongoize(nan)).to eq("NaN")
+        end
       end
 
-      it "returns a BigDecimal" do
-        expect(demongoized).to be_a(BigDecimal)
+      context "when the value is BigDecimal Infinity" do
+
+        let(:infinity) do
+          BigDecimal("Infinity")
+        end
+
+        it "returns a String" do
+          expect(BigDecimal.mongoize(infinity)).to eq("Infinity")
+        end
       end
 
-      it "is a negative infinity BigDecimal" do
-        expect(demongoized.infinite?).to eq -1
+      context "when the value is BigDecimal negative Infinity" do
+
+        let(:neg_infinity) do
+          BigDecimal("-Infinity")
+        end
+
+        it "returns a String" do
+          expect(BigDecimal.mongoize(neg_infinity)).to eq("-Infinity")
+        end
       end
-    end
-  end
 
-  describe ".mongoize" do
+      context "when the value is the constant Float::NAN" do
 
-    context "when the value is a BigDecimal" do
+        let(:nan) do
+          Float::NAN
+        end
 
-      it "returns a string" do
-        expect(BigDecimal.mongoize(big_decimal)).to eq(big_decimal.to_s)
+        it "returns a String" do
+          expect(BigDecimal.mongoize(nan)).to eq("NaN")
+        end
       end
-    end
 
-    context "when the value is the BigDecimal zero" do
+      context "when the value is constant Float::INFINITY" do
+
+        let(:infinity) do
+          Float::INFINITY
+        end
 
-      let(:big_decimal) do
-        BigDecimal("0.0")
+        it "returns a String" do
+          expect(BigDecimal.mongoize(infinity)).to eq("Infinity")
+        end
       end
 
-      it "returns a BigDecimal" do
-        expect(BigDecimal.mongoize(big_decimal)).to eq(big_decimal.to_s)
+      context "when the value is constant Float::INFINITY * -1" do
+
+        let(:neg_infinity) do
+          Float::INFINITY * -1
+        end
+
+        it "returns a String" do
+          expect(BigDecimal.mongoize(neg_infinity)).to eq("-Infinity")
+        end
       end
     end
 
-    context "when the value is the BigDecimal negative zero" do
+    describe "#mongoize" do
 
-      let(:big_decimal) do
-        BigDecimal("-0.0")
+      it "returns a string" do
+        expect(big_decimal.mongoize).to eq(big_decimal.to_s)
       end
+    end
+
+    describe "#numeric?" do
 
-      it "returns a BigDecimal" do
-        expect(BigDecimal.mongoize(big_decimal)).to eq(big_decimal.to_s)
+      it "returns true" do
+        expect(big_decimal.numeric?).to eq(true)
       end
     end
+  end
+
+  context 'when map_big_decimal_to_decimal128 is true' do
+
+    before do
+      Mongoid.map_big_decimal_to_decimal128 = true
+    end
+
+    after do
+      Mongoid.map_big_decimal_to_decimal128 = false
```

Comment:
I would prefer an `around` block instead which would read the current setting value and restore the setting to the current value, instead of hardcoding our present default like this.

Reply:
done